File: 1071-BinaryPrefixDivisibleBy5/1071-BinaryPrefixDivisibleBy5.py
# Last updated: 2/10/2026, 8:47:43 PM
class Solution(object):
    def prefixesDivBy5(self, nums):
      string=''
      lis=[]
      for i in nums:
         string+=str(i)
         y=int(string,2)
         if y%5==0:
            lis.append(True)
         else:
            lis.append(False)
      return lis

File: 1071-BinaryPrefixDivisibleBy5/1071-BinaryPrefixDivisibleBy5.cpp
// Last updated: 7/7/2026, 10:25:12 PM
class Solution {
public:
    vector<bool> prefixesDivBy5(vector<int>& nums) {
        vector<bool> ans;
        int val = 0;
        for (int bit : nums) {
            val = (val * 2 + bit) % 5;
            ans.push_back(val == 0);
        }
        return ans;
    }
};